Tests/SwiftAlgorithmsTests/TestUtilities.swift

```diff
@@ -100,15 +110,59 @@ func XCTAssertUnorderedEqualSequences<S1: Sequence, S2: Sequence>(
     s1.remove(at: idx)
```

Comment:
When it was implemented perf was not in mind because well it was just testing. There is maybe a small optimization that can be done here as well.
```suggestion
    // Since ordering does not matter, this is a small
    // optmization that avoid removing a index in the middle
    // of the array and having to slide all elements.
    let lastIdx = s1.index(before: s1.endIndex)
    if lastIdx != idx {
      s1.swapAt(idx, lastIdx)
    }
    s1.remove(at: lastIdx)
```
So by swapping always to back and removing there, we avoid removal in middle which my guess need to slide all the elements after one position after removal. 

Didn't test it, is just a guess, but throwing the idea here in case you want to give it a shot since you are measuring as well =]

Reply:
The logic makes sense to me. I did a quick benchmark on two random arrays of integers and there is consistent performance improvements in larger collections. This is an interesting use case for an unordered collection, like `Set` but without uniqueness or the `Hashable` requirements.

<img width="1280" alt="chart" src="https://user-images.githubusercontent.com/963948/147159107-91eabff9-c6c2-4d10-9271-f681c8623865.png">

Reply:
I updated the documentation and removed the need for the redundant `!=` check since that already occurs internally in [`swapAt(_:_:)`](https://github.com/apple/swift/blob/59559ca0ce668b0240bcb2a91f874922a8eba2d7/stdlib/public/core/MutableCollection.swift#L321).

Reply:
Awesome! Thanks for the measurements :)